### PR TITLE
feat!: renames `react-dropdowns` to `react-dropdowns.legacy`

### PR DESCRIPTION
**Pull request by geotrev**
## Description

Renames `react-dropdowns` and associated metadata throughout the repo to `react-dropdowns.legacy`.

## Checklist

- ~~:ok_hand: design updates will be Garden Designer approved (add the designer as a reviewer)~~
- [x] :globe_with_meridians: demo is up-to-date (`npm start`)
- ~~:arrow_left: renders as expected with reversed (RTL) direction~~
- ~~:metal: renders as expected with Bedrock CSS (`?bedrock`)~~
- ~~:guardsman: includes new unit tests. Maintain existing coverage (always >= 96%)~~
- ~~:wheelchair: tested for WCAG 2.1 AA accessibility compliance~~
- ~~:memo: tested in Chrome, Firefox, Safari, and Edge~~

### CODE REVIEW

File: docs/migration.md [hunk 1]
```diff
@@ -60,6 +60,8 @@ consider additional positioning prop support on a case-by-case basis.
 
 #### @zendeskgarden/react-dropdowns
 
+- The `v8` version of this package is no longer maintained and is
+  renamed to `@zendeskgarden/react-dropdowns.legacy` in `v9`
```
Comment (geotrev) [inline docs/migration.md:64]: Open to ideas on if a better location in this file or clearer description would be useful.

File: packages/notifications/demo/stories/ToastStory.tsx [hunk 2]
```diff
@@ -20,7 +20,7 @@ import {
   MediaItem,
   Menu,
   Trigger
-} from '@zendeskgarden/react-dropdowns';
+} from '@zendeskgarden/react-dropdowns.legacy';
```
Comment (jzempel) [inline packages/notifications/demo/stories/ToastStory.tsx:23]: This story should be using the non-legacy `Menu`.

File: docs/migration.md (hunk 1 above)
Comment (jzempel) [reply docs/migration.md:64]: Seems as good as anything to me.

**Review by jzempel — CHANGES_REQUESTED** [review]

File: packages/notifications/demo/stories/ToastStory.tsx (hunk 2 above)
Comment (geotrev) [reply packages/notifications/demo/stories/ToastStory.tsx:23]: Yep, I realized that last night after putting up the PR. On it now :)

**Review by jzempel — APPROVED** [review]
ah, that new `Menu` API is so pleasing 🙂

**Pull request merged**